src/platforms/esp32/components/avm_builtins/uart_driver.c

```diff
@@ -62,8 +62,8 @@ struct UARTData
 {
     QueueHandle_t rxqueue;
     EventListener listener;
-    term reader_process_pid;
-    uint64_t reader_ref_ticks;
+    _Atomic term reader_process_pid;
+    _Atomic uint64_t reader_ref_ticks;
```

Comment:
Now I have some doubts about this fix, it seems like the `reader_process_pid` and `reader_ref_ticks` could get out of sync, if a process were to change the `reader_process_id`, while another was waiting to change it, and the first process was interrupted by a higher priority task (before changing `reader_ref_ticks`), and the second then was allowed to change `reader_process_id` and then `reader_ref_ticks`, before the first process was scheduled to resume.

Reply:
Or both?